core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -382,11 +392,37 @@ private[spark] class AppStatusListener(
       }
 
       job.status = event.jobResult match {
-        case JobSucceeded => JobExecutionStatus.SUCCEEDED
-        case JobFailed(_) => JobExecutionStatus.FAILED
+        case JobSucceeded =>
+          appStatusSource.foreach{_.SUCCEEDED_JOBS.inc(1)}
+          JobExecutionStatus.SUCCEEDED
+        case JobFailed(_) =>
+          appStatusSource.foreach{_.FAILED_JOBS.inc(1)}
+          JobExecutionStatus.FAILED
       }
 
       job.completionTime = if (event.time > 0) Some(new Date(event.time)) else None
+
+      for {
+        submissionTime <- job.submissionTime
+        completionTime <- job.completionTime
+      } yield {
```

Comment:
Nit: don't you technically need `foreach` here? I think it might be clearer to just wrap this in `if (job.submissionTime.isDefined && ...)` rather than the `for yield` that doesn't really loop.
Below -- isn't this just trying to take the difference in time in milliseconds between two dates? if so don't you just subtract their timestamps or is it more subtle here?

Reply:
Yeah I wanted to do this properly (java 8 wise, didnt think much of it) so on one hand I do have the timestamp on the other hand I do have the date object. So it was straightforward to pick the date. I didnt extract a timestamp out of the date object.

Reply:
`java.util.Date.getTime` gives you ms since the epoch directly, so I think you can subtract them and be done?

Reply:
It should be possible just tried the elegant java way: https://stackoverflow.com/questions/4927856/how-to-calculate-time-difference-in-java :)

Reply:
@srowen regarding multiple options its a valid technique: https://alvinalexander.com/scala/how-to-use-multiple-options-for-loop-comprehension
Anyway I can keep it simple and  just use ifDefined and then get the values, it liked the former though, more idiomatic.